tests/test_schema.py

```diff
@@ -406,3 +406,9 @@ class Model(BaseModel):
             },
         },
     }
+
+
+def test_field_name_shadows_attribute():
```

Comment:
this is the wrong place, I guess move to `test_main.py`. You should also add a test for a non annotation only attribute.